src/pathfora.js

```diff
@@ -505,14 +505,49 @@
       return true;
     },
 
-    pageVisitsChecker: function (pageVisitsRequired, widget) {
-      return (core.pageViews >= pageVisitsRequired);
+    /**
+     * @description Parse url queries as an object
+     * @param {string} url
+     */
+    parseQueries: function (url) {
+      var queries = {};
+      var pieces = url.split('?');
+      if (pieces.length > 1) {
+        pieces = pieces[1].split('&');
+
+        for (var i = 0; i < pieces.length; i++) {
+          var pair = pieces[i].split('=');
+
+          if (pair.length > 1) {
+            queries[pair[0]] = pair[1];
+          }
+        }
+      }
+      return queries;
     },
 
+    /**
+     * @description Compare query params between the url
+     *              the user is visiting and the match
+     *              rule provided
+     * @param {obj} queries
+     * @param {obj} matchQueries
+     */
+    compareQueries: function (queries, matchQueries) {
+      // any query in the matchQueries should exist in the url in the browser
+      for (var key in matchQueries) {
+        if (matchQueries[key] !== queries[key]) {
+          return false;
+        }
+      }
+
+      return true;
+    },
 
     urlChecker: function (phrases, widget) {
-      var url = window.location.href,
-          simpleurl = window.location.hostname + window.location.pathname,
+      var url = decodeURIComponent(window.location.href),
```

Comment:
whats the reason for `decodeURIComponent` here (or anywhere in the urls you are comparing)?

if my url contains encoded params it could cause things to break in a variety of ways. for example:
`https://www.getlytics.com/?bonk=%7B%22one%22%3A%22two%3F%22%7D`
is valid but decodes to
`https://www.getlytics.com/?bonk={"one":"two?"}`
which will break when you split on `?`